common/src/stack/switch/command/list/switch/partition/member/__init__.py

```diff
@@ -93,9 +99,18 @@ def run(self, params, args):
 		member_select += ' ORDER BY switch, host, part_name'
 
 		self.beginOutput()
-		for line in self.db.select(member_select, vals):
-			if expanded:
-				line = list(line)
-				line[6] = '0x{0:04x}'.format(line[6])
-			self.addOutput(line[0], (line[1:]))
+
+		# Get the information from Stacki's database
+		if source == 'database':
+			for line in self.db.select(member_select, vals):
+				if expanded:
+					line = list(line)
+					line[6] = '0x{0:04x}'.format(line[6])
+				self.addOutput(line[0], (line[1:]))
+
+		# Get the information from the switch itself if source is set to switch
+		elif source == 'switch':
+			for switch_name in switches:
+				model = self.getHostAttr(switch_name, 'component.model')
+				self.runImplementation(model, [switch_name, expanded])
```

Comment:
I'm surprised that the implementation is calling `addOutput` because I think the way I've written implementations is to have them return the raw results so that the higher layer invoking them can decide how to format it. I don't know if this is the standard, but I think it would be a good idea to not have the implementation dictate how the output must be used (in this case forcing the issue by using addOutput in the implementation).

Reply:
That's a good point, if you look at the implementation for `list switch status` you can see the pattern I was basing my implementation from. I also see the same pattern used in `report switch` for the x1052 implementation as well. Whether this is the right approach or not I'm not sure of, but it has been used before.

Reply:
Git blame tells me to blame @bsanders, so I'll defer to him on this one.